### ISSUE TITLE
you have done an excellent code but i am facing some issues while reproducing your paper. i am using the packages and libraries that you mentioned in requirement.txt

### ISSUE THREAD

**Issue by ikraj** [vqdang/hover_net#38]

**Comment by ikraj** [comment]
![git](https://user-images.githubusercontent.com/63111211/78373560-32976f80-75e4-11ea-9b4a-9bd5bfbcd989.png)
iam facing these errors while running train.py file. guide me to resolve these issues.

**Comment by vqdang** [comment]
Hi, likely this is due to mis-match version of tensorpack. Could you please create a fresh environment using the following and rerun again?

```
conda create --name test python=3.6
conda activate test
pip install opencv-python=3.2 scipy scikit-image pandas matplotlib
pip install --upgrade git+https://github.com/tensorpack/tensorpack.git@0.9.0.1
pip install --upgrade https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow_gpu-1.12.0-cp36-cp36m-linux_x86_64.whl
source activate test
```

**Comment by vqdang** [comment]
Close due to inactivity

**Comment by ikraj** [comment]
sorry, but can we reopen this?
because i have created a fresh environment as you mentioned but getting the same error.

**Comment by vqdang** [comment]
Please, detail as much as possible what you have done. 
1. Dataset. Is it your own, or the one we provided i.e CoNSeP?
2. Any modifications within the source code.

Also, please read this for self-diagnose https://github.com/vqdang/hover_net/issues/25

**Comment by ikraj** [comment]
iam using dataset of CoNSep and no modification other thn the path for the pretrained file.

**Comment by vqdang** [comment]
Did you run the `extract_patch.py` to generate training / validation image patches ?

**Comment by simongraham** [comment]
In line with what @vqdang has mentioned regarding `extract_patches.py` can you please describe the data format of your files. Have you saved individual numpy files per patch? Also please let us know what the dimensions of each numpy file are. Each patch should be `HxWx2` where `H` and `W` are the spatial dimensions of the patch (we use 270x270 for HoVer-Net). Then the first channel should be the instance segmentation map and the second channel should be a map denoting the class of each pixel. If you have not organised your data in this format-you will likely get an error.

**Comment by ikraj** [comment]
as you say i run the file extract_patches.py and get folder named 540x540_80x80 and the files in thee folder are .npy  format. and yes numpy file for each patch is generated.  now iam getting
![asser](https://user-images.githubusercontent.com/63111211/79043000-ac97ac00-7c15-11ea-82f4-608b7da810da.png)
 the error while running python3 train.py --gpu='0,1' that mention in attachement. for this error i already read the issue 12 but could not resolve the issue. and my other question is that you mention path for validation set directory but there is no validation set so which images i should used for the validation set.  
thank you.

**Comment by simongraham** [comment]
This error means that the paths that you are giving to the train / validation directories are incorrect. If, as you have suggested, you haven’t extracted validation patches this will also explain the error. You can devise a suitable validation set from the training set and extract the patches in the same way as the training set.

**Comment by ikraj** [comment]
the error remains the same after extracting validation patches. these are the path of validation and train set directories.
![dir](https://user-images.githubusercontent.com/63111211/79049749-e0d39280-7c3e-11ea-8cdd-59f9c4e1139d.png)

**Comment by vqdang** [comment]
If you run `extract_patches.py` without modifying anything besides the path, I have a feeling you are missing a dash after `train` here. The provided path in the current `config.py` is as below:
`../../CoNSeP/train/%s/'  % data_code_dict[self.model_type]`
`------------------^----` The paths in your screenshot miss this.

**Comment by ikraj** [comment]
please check my directories because errors is still there.
![assrtled](https://user-images.githubusercontent.com/63111211/79053330-fe602680-7c55-11ea-8902-26ed24ceed7b.png)

**Comment by vqdang** [comment]
I'm not talking about `extract_patches.py`. I'm talking about this. You have to add the `/` before `%s`
![79049749-e0d39280-7c3e-11ea-8cdd-59f9c4e1139d](https://user-images.githubusercontent.com/24943262/79057066-e340e700-7c54-11ea-9b9e-fac3b1a6f0a0.png)
And also, from the 2nd screenshot, you provided the wrong path here. Your folder is `Train`, not `train`. You have to point the paths in the `config.py` correctly.
![79053330-fe602680-7c55-11ea-8902-26ed24ceed7b](https://user-images.githubusercontent.com/24943262/79057098-3a46bc00-7c55-11ea-944e-321a3bb27f01.png)

**Comment by vqdang** [comment]
Close due to inactivity
